app/controllers/users/passwords_controller.rb

```diff
@@ -39,11 +43,12 @@ def capture_password_if_pii_present_but_locked
       redirect_to capture_password_url
     end
 
-    def user_params
-      params.require(:update_user_password_form).permit(:password, :password_confirmation)
+    def required_password_change
+      @required_password_change ||= session[:redirect_to_change_password]
```

Comment:
1. Session access is a pretty trivial operation, so memoization seems unnecessary
2. Memoizing a boolean value this way doesn't actually memoize `false` values, since it'll repeatedly try reassigning
3. I'd expect a `?` suffix for a predicate method. Although technically this could also return `nil`, maybe we want to coerce to a boolean to limit return types

```suggestion
    def required_password_change?
      session[:redirect_to_change_password] == true
```